packages/typespec-client-generator-core/src/package.ts

```diff
@@ -660,7 +660,16 @@ function addDefaultClientParameters<
   if (apiVersionParam) {
     client.initialization.properties.push(apiVersionParam);
   }
-  const subId = client.initialization.properties.find((x) => isSubscriptionId(context, x));
+  let subId = context.__clientToParameters.get(client.__raw.type)?.find((x) => isSubscriptionId(context, x));
+  if (!subId && context.arm) {
+    for (const operationGroup of listOperationGroups(context, client.__raw)) {
+      // if any sub operation groups have an subId param, the top level needs it as well
+      subId = context.__clientToParameters
+        .get(operationGroup.type)
+        ?.find((x) => x.isApiVersionParam);
+      if (apiVersionParam) break;
+    }
+  }
```

Comment:
`subscriptionId` is special parameter in ARM SDK and we shall handle it with similar logic as `api-verison`.

Reply:
updated